server/sonar-server/src/main/java/org/sonar/server/component/ComponentUpdater.java

```diff
@@ -100,6 +103,17 @@ private ComponentDto createRootComponent(DbSession session, NewComponent newComp
       .setPrivate(newComponent.isPrivate())
       .setCreatedAt(new Date(system2.now()));
     dbClient.componentDao().insert(session, component);
+
+    BranchDto branch = new BranchDto()
```

Comment:
it needs to be created as soon as the project is created if we want it to always be consistent. I will change it to only be done when it's a regular project.